javascript/tool/js/housing-insights.js

```diff
@@ -46,9 +46,97 @@ String.prototype.hashCode = function() {
   return 'd' + hash;
 };
 
+// This constructor stores json data fetched from AWS so that we can do various things with it,
+// like convert it to geoJSON or grab its display name and other information 
+// (regardless of the data source) using conventions from meta.json.
+// This assumes that 'json' is an array of objects.
+function APIDataObj(json){
+  this.json = json;
+  
+  // this.geoJSON assumes that we'll be producing a FeatureCollection with Features
+  // that are points. For shapes, we may want to add an argument and code that responds to it.
+  // Currently this method requires us to specify the latitude and longitude fields
+  // of the data within the json. If we standardize the json to always call its geospatial
+  // fields 'longitude' and 'latitude', this may not be necessary.
+  this.geoJSON = function(longitudeField, latitudeField){
+    var features = json.map(function(element){
+      return {
+        'type': 'Feature',
+        'geometry': {
+          'type': 'Point',
+          'coordinates': [+element[longitudeField], +element[latitudeField]]
+        },
+        'properties': element        
+      }
+    });
+    console.log('json', json);
+    return {
+      'type': 'FeatureCollection',
+      'features': features
+    }
+  }
+  
+}
+
 
 var app = {
     dataCollection: {}, // empty object to house potentially shared data called by specific charts, see above
+    
+    // getAPIData exists to fetch data from the AWS API based on meta.json. This is a separate 
```

Comment:
I think this sounds like a great approach - keep both methods available side by side as we transition. I don't see a problem with keeping it named getAPIdata even after getData is deprecated. 

I tested this locally and worked. I am going to modify this to instead use the Flask API demo after merging.